### PR TITLE
feat: Add new Keybind option "Select all navy, without mobilesonar"/ make the original beavior consistent #4459

### PR DESCRIPTION
**Pull request by xXEddieXxx**
## Description of the changes
I added a new Keybind option :"select_naval_no_sonar". 
It selects all navy units except sonar platforms (Ships with sonar trait still get selected, mabye the naming needs to be refactored).

I also changed the t2 seraphim sonar platform that actually was not selected when the select Navy keybind was used.(in contrast to all other mobile sonars)
It is now consisten with all sonarplatfroms that when u select all navy they also get selected. 

## Test setup for the changes
Spawn in as many navy units and sonarplatforms as you like and try the new Keybindoption. 

## Additional context
If there is anything off with description or naming just tell me i will change it :)

### CODE REVIEW

**Comment by 4z0t** [comment]
> @4z0t is there no way to subtract categories using the console command `UI_SelectByCategory`?

@Garanas no, only union and intersection.

**Review by Garanas — APPROVED** [review]

File: lua/keymap/keydescriptions.lua [hunk 1]
```diff
@@ -326,6 +326,7 @@ keyDescriptions = {
     ['select_air_no_transport'] = '<LOC key_desc_0253>Select air but no transports',
     ['select_air_transport'] = '<LOC key_desc_0254>Select only transports',
     ['lock_zoom'] = '<LOC key_desc_0255>Lock the zoom to current level',
+    ['select_naval_no_sonar'] = 'Select all Naval units except Sonarplatforms',
```
Comment (Garanas) [inline lua/keymap/keydescriptions.lua:329]: Is 'Sonarplatforms' one word?

Comment (xXEddieXxx) [reply lua/keymap/keydescriptions.lua:329]: I will check it after work and change it acordingly. :D

File: lua/keymap/misckeyactions.lua [hunk 2]
```diff
@@ -17,7 +17,9 @@ function lockZoom()
     end
 end
 
-
+function navalNoSonar()
+    UISelectionByCategory("NAVAL + MOBILE", false, false, false, false) SelectUnits(EntityCategoryFilterDown(categories.ALLUNITS - categories.MOBILESONAR, GetSelectedUnits()))
```
Comment (4z0t) [inline lua/keymap/misckeyactions.lua:21]: This plus doesn't do anything actually, remove it to not confuse anyone.

**Review by 4z0t — CHANGES_REQUESTED** [review]

File: lua/keymap/misckeyactions.lua (hunk 2 above)
[the remark this answers is not in the document]
Comment (xXEddieXxx) [reply lua/keymap/misckeyactions.lua:21]: Can you explain ?

[the remark this answers is not in the document]
Comment (Garanas) [reply lua/keymap/misckeyactions.lua:21]: There's no need to cache this. The code path is not critical.

@xXEddieXxx this is caching of catories:
- https://github.com/FAForever/fa/blob/deploy/fafdevelop/lua/defaultunits.lua#L25-L27

It prevents re-computing the categories arithmetic's. Those arithmetic's are engine calls and are therefore expensive. Caching the computation is useful when you're dealing with critical paths (paths that are run so often that you want to get everything out of it that you can possibly get, performance-wise)

Comment (Garanas) [reply lua/keymap/misckeyactions.lua:21]: If you happen to run into it - yes. But don't go looking for it 😄 !

Comment (xXEddieXxx) [reply lua/keymap/misckeyactions.lua:21]: Love that sentence ! xD

File: lua/keymap/keydescriptions.lua (hunk 1 above)
Comment (Garanas) [reply lua/keymap/keydescriptions.lua:329]: image

I think it is two separate words. Could you adjust that? After that we can merge.

**Comment by xXEddieXxx** [comment]
ok about my last PR i investigated a little bit more and found, that someone wrote a smart select(wich is used rarely), wich gives you the opportunity to use a Keybind and give it string attriubutes also substraction as you mentioned @Garanas. 
So i deleted my own function and used this instead, i dont know if this is better but it looked nicer to me then having a special function for each Keybind, that are all doing the same.(like my navalnosonar function or the Airnotransport function)
I would like your thoughts about this aswell @4z0t the function looked pretty old can this be used ?

**Comment by Garanas** [comment]
I'm not a huge fan of smart selection because of the manual parsing. But for the sake of a hotkey - it is fine 😃

**Review by Garanas — APPROVED** [review]

**Pull request merged**